tool/live-monitor.js

```diff
@@ -0,0 +1,166 @@
+
+const defaultCommandsValues = {
+    fromBlock: 'latest',
+    methods: [],
+    events: [],
+    pegout: false,
+    pegin: false,
+    network: 'testnet',
+    checkEveryMillis: DEFAULT_CHECK_EVERY_MILLIS,
+};
+
+const getParsedCommands = () => {
+    const commands = process.argv.filter(command => command.startsWith('--'))
+    .reduce((commands, command) => {
+        if(command.startsWith('--fromblock')) {
+            commands.fromBlock = command.slice(command.indexOf('=') + 1);
+        } else if(command.startsWith('--methods') || command.startsWith('--events')) { // Parsing commands that include an array
+            commands['methods'] = JSON.parse(command.slice(command.indexOf('=') + 1).replaceAll("'", '"'));
+        } else if(command.startsWith('--network')) {
+            commands.network = command.slice(command.indexOf('=') + 1);
+        } else if(command.startsWith('--pegout') || command.startsWith('--pegin')) {
+            const parsedCommand = command.replace('--', '');
+            commands[parsedCommand] = true;
+        } else if(command.startsWith('--checkeverymillis')) {
+            commands.checkEveryMillis = Number(command.slice(command.indexOf('=') + 1));
+        } else {
+            throw new Error(`Command '${command}' not recognized`);
+        }
+        return commands;
+    }, defaultCommandsValues);
+    return commands;
+};
+
+const PEGOUT_METHOD_SIGNATURES = {
+    releaseBtc: '0x',
+    updateCollections: '0x0c5a9990',
+    addSignature: '0xf10b9c59'
+};
+
+const PEGIN_METHOD_SIGNATURES = {
+    registerBtcCoinbaseTransaction: '0xccf417ae',
+    registerBtcTransaction: '0x43dc0656',
+};
+
+function isAPegoutRelatedTransactionData(data) {
+    const methodSignature = data.slice(0, 10);
+    switch(methodSignature) {
+        case PEGOUT_METHOD_SIGNATURES.releaseBtc:
+        case PEGOUT_METHOD_SIGNATURES.updateCollections:
+        case PEGOUT_METHOD_SIGNATURES.addSignature:
+            return true;
+        default:
+            return false;
+    }
+}
+
+function isAPeginRelatedTransactionData(data) {
+    const methodSignature = data.slice(0, 10);
+    switch(methodSignature) {
+        case PEGIN_METHOD_SIGNATURES.registerBtcTransaction:
+            return true;
+        default:
+            return false;
+    }
+}
+
+const commands = getParsedCommands();
+
+const web3Client = new Web3(networkParser(commands.network));
+
+let currentBlockNumber = commands.fromBlock;
+
+let notified = false;
+
+async function monitor() {
+
+    try {
+
+        const latestBlockNumber = await web3Client.eth.getBlockNumber();
+
+        if(currentBlockNumber === 'latest') {
+            currentBlockNumber = latestBlockNumber;
+        }
+
+        if(latestBlockNumber < currentBlockNumber) {
+            if(!notified) {
+                console.warn("latestBlockNumber was reached!");
+                notified = true;
+            }
+            return;
+        }
+
+        notified = false;
```

Comment:
Do you need to set it again?

Reply:
Yes.
This is used to avoid printing the message `"latestBlockNumber was reached!"` many times when the latest block hasn't changed
If we are checking every second, and we are already in the latest block, then we would be printing `"latestBlockNumber was reached!"` many times for the same block. So, with this `notified` flag I only show that message when there is a new best block that we found.

Reply:
Makes sense. Thanks!